### PR TITLE
cognito identity pool - refactor to use keyvaluetags support

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates #10688

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
NONE
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSCognitoIdentityPool_'
=== RUN   TestAccAWSCognitoIdentityPool_basic
=== PAUSE TestAccAWSCognitoIdentityPool_basic
=== CONT  TestAccAWSCognitoIdentityPool_basic
--- PASS: TestAccAWSCognitoIdentityPool_basic (59.44s)
=== RUN   TestAccAWSCognitoIdentityPool_supportedLoginProviders
=== PAUSE TestAccAWSCognitoIdentityPool_supportedLoginProviders
=== CONT  TestAccAWSCognitoIdentityPool_supportedLoginProviders
--- PASS: TestAccAWSCognitoIdentityPool_supportedLoginProviders (83.04s)
=== RUN   TestAccAWSCognitoIdentityPool_openidConnectProviderArns
=== PAUSE TestAccAWSCognitoIdentityPool_openidConnectProviderArns
=== CONT  TestAccAWSCognitoIdentityPool_openidConnectProviderArns
--- FAIL: TestAccAWSCognitoIdentityPool_openidConnectProviderArns (51.52s)
    testing.go:635: Step 2 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: aws_cognito_identity_pool.main
          allow_unauthenticated_identities: "false" => "false"
          arn:                              "arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1" => "arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1"
          cognito_identity_providers.#:     "0" => "0"
          developer_provider_name:          "" => ""
          id:                               "us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1" => "us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1"
          identity_pool_name:               "identity pool cz10v1q896" => "identity pool cz10v1q896"
          openid_connect_provider_arns.#:   "2" => "2"
          openid_connect_provider_arns.0:   "arn:aws:iam::123456789012:oidc-provider/bar.example.com" => "arn:aws:iam::123456789012:oidc-provider/foo.example.com"
          openid_connect_provider_arns.1:   "arn:aws:iam::123456789012:oidc-provider/foo.example.com" => "arn:aws:iam::123456789012:oidc-provider/bar.example.com"
          saml_provider_arns.#:             "0" => "0"
        
        
        
        STATE:
        
        aws_cognito_identity_pool.main:
          ID = us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1
          provider = provider.aws
          allow_unauthenticated_identities = false
          arn = arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:83701026-87bb-49bc-ad27-864433cd5fa1
          developer_provider_name = 
          identity_pool_name = identity pool cz10v1q896
          openid_connect_provider_arns.# = 2
          openid_connect_provider_arns.0 = arn:aws:iam::123456789012:oidc-provider/bar.example.com
          openid_connect_provider_arns.1 = arn:aws:iam::123456789012:oidc-provider/foo.example.com
=== RUN   TestAccAWSCognitoIdentityPool_samlProviderArns
=== PAUSE TestAccAWSCognitoIdentityPool_samlProviderArns
=== CONT  TestAccAWSCognitoIdentityPool_samlProviderArns
--- PASS: TestAccAWSCognitoIdentityPool_samlProviderArns (89.81s)
=== RUN   TestAccAWSCognitoIdentityPool_cognitoIdentityProviders
=== PAUSE TestAccAWSCognitoIdentityPool_cognitoIdentityProviders
=== CONT  TestAccAWSCognitoIdentityPool_cognitoIdentityProviders
--- PASS: TestAccAWSCognitoIdentityPool_cognitoIdentityProviders (85.06s)
=== RUN   TestAccAWSCognitoIdentityPool_addingNewProviderKeepsOldProvider
--- PASS: TestAccAWSCognitoIdentityPool_addingNewProviderKeepsOldProvider (83.56s)
=== RUN   TestAccAWSCognitoIdentityPool_tags
=== PAUSE TestAccAWSCognitoIdentityPool_tags
=== CONT  TestAccAWSCognitoIdentityPool_tags
--- PASS: TestAccAWSCognitoIdentityPool_tags (84.66s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-aws/aws	233.111s

...
```

the error seems unrelated to the change

### CODE REVIEW

File: aws/resource_aws_cognito_identity_pool_test.go [hunk 1]
```diff
@@ -29,8 +29,9 @@ func TestAccAWSCognitoIdentityPool_basic(t *testing.T) {
 				Check: resource.ComposeAggregateTestCheckFunc(
 					testAccCheckAWSCognitoIdentityPoolExists(resourceName),
 					resource.TestCheckResourceAttr(resourceName, "identity_pool_name", fmt.Sprintf("identity pool %s", name)),
-					resource.TestMatchResourceAttr(resourceName, "arn",
-						regexp.MustCompile("^arn:aws:cognito-identity:[^:]+:[0-9]{12}:identitypool/[^:]+:([0-9a-f]){8}-([0-9a-f]){4}-([0-9a-f]){4}-([0-9a-f]){4}-([0-9a-f]){12}$")),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "cognito-identity", regexp.MustCompile(`identitypool/.+`)),
+					//resource.TestMatchResourceAttr(resourceName, "arn",
+					//	regexp.MustCompile("^arn:aws:cognito-identity:[^:]+:[0-9]{12}:identitypool/[^:]+:([0-9a-f]){8}-([0-9a-f]){4}-([0-9a-f]){4}-([0-9a-f]){4}-([0-9a-f]){12}$")),
```
Comment (bflad) [inline aws/resource_aws_cognito_identity_pool_test.go]: Commented code can be removed 👍

Comment (DrFaust92) [reply aws/resource_aws_cognito_identity_pool_test.go]: fixed

File: aws/resource_aws_cognito_identity_pool.go [hunk 2]
```diff
@@ -199,6 +200,15 @@ func resourceAwsCognitoIdentityPoolUpdate(d *schema.ResourceData, meta interface
...
 
+	arn := d.Get("arn").(string)
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+
+		if err := keyvaluetags.CognitoidentityUpdateTags(conn, arn, o, n); err != nil {
+			return fmt.Errorf("error updating Cognito Identity Pool (%s) tags: %s", arn, err)
+		}
+	}
+
```
[the remark this answers is not in the document]
Comment (DrFaust92) [reply aws/resource_aws_cognito_identity_pool.go]: d'oh!
fixed, sorry about that.

**Comment by DrFaust92** [comment]
@bflad, fixed based on comments, sorry again,

```
--- PASS: TestAccAWSCognitoIdentityPool_basic (66.22s)
--- PASS: TestAccAWSCognitoIdentityPool_supportedLoginProviders (138.92s)
--- FAIL: TestAccAWSCognitoIdentityPool_openidConnectProviderArns (59.22s)
    testing.go:635: Step 2 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: aws_cognito_identity_pool.main
          allow_unauthenticated_identities: "false" => "false"
          arn:                              "arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:be33d615-5d36-4185-9483-d561a81c25a8" => "arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:be33d615-5d36-4185-9483-d561a81c25a8"
          cognito_identity_providers.#:     "0" => "0"
          developer_provider_name:          "" => ""
          id:                               "us-west-2:be33d615-5d36-4185-9483-d561a81c25a8" => "us-west-2:be33d615-5d36-4185-9483-d561a81c25a8"
          identity_pool_name:               "identity pool 70x99xj2ns" => "identity pool 70x99xj2ns"
          openid_connect_provider_arns.#:   "2" => "2"
          openid_connect_provider_arns.0:   "arn:aws:iam::123456789012:oidc-provider/bar.example.com" => "arn:aws:iam::123456789012:oidc-provider/foo.example.com"
          openid_connect_provider_arns.1:   "arn:aws:iam::123456789012:oidc-provider/foo.example.com" => "arn:aws:iam::123456789012:oidc-provider/bar.example.com"
          saml_provider_arns.#:             "0" => "0"
        
        
        
        STATE:
        
        aws_cognito_identity_pool.main:
          ID = us-west-2:be33d615-5d36-4185-9483-d561a81c25a8
          provider = provider.aws
          allow_unauthenticated_identities = false
          arn = arn:aws:cognito-identity:us-west-2:476956259333:identitypool/us-west-2:be33d615-5d36-4185-9483-d561a81c25a8
          developer_provider_name = 
          identity_pool_name = identity pool 70x99xj2ns
          openid_connect_provider_arns.# = 2
          openid_connect_provider_arns.0 = arn:aws:iam::123456789012:oidc-provider/bar.example.com
          openid_connect_provider_arns.1 = arn:aws:iam::123456789012:oidc-provider/foo.example.com
--- PASS: TestAccAWSCognitoIdentityPool_samlProviderArns (98.48s)
--- PASS: TestAccAWSCognitoIdentityPool_cognitoIdentityProviders (92.42s)
--- PASS: TestAccAWSCognitoIdentityPool_addingNewProviderKeepsOldProvider (216.50s)
--- PASS: TestAccAWSCognitoIdentityPool_tags (113.69s)

```

**Pull request merged**